components/org.wso2.carbon.identity.application.authenticator.adapter/src/main/java/org/wso2/carbon/identity/application/authenticator/adapter/AuthenticationResponseProcessor.java

```diff
@@ -0,0 +1,244 @@
+                break;
+            default:
+                break;
+        }
+    }
+
+    /**
+     * Only the REPLACE operation is allowed. If any other operation is provided, it will be ignored
+     * and the authenticated user will not be updated.
+     *
+     * @param operation Operation from the external authentication service response.
+     */
+    private boolean isAllowedOperation(Operation operation) {
+
+        return Operation.REPLACE.equals(operation);
+    }
+
+    /**
+     * Create a new authenticated user based on the executed operations.
+     *
+     * @throws ActionExecutionResponseProcessorException    If error occurred while creating authenticated user
+     *                                                      due to invalid operation.
+     * @throws UserStoreException                           If error occurred while retrieving local user details.
+     */
+    public AuthenticatedUser createAuthenticateduser(AuthenticationRequestUser user, AuthenticationContext context,
+            UserStore userStore) throws ActionExecutionResponseProcessorException, UserStoreException {
+
+        AuthenticatedUser authenticatedUser = new AuthenticatedUser();
+        authenticatedUser.setFederatedUser(user.resolveUserType());
+        authenticatedUser.setUserAttributes(user.resolveUserClaims());
+
+        if (authenticatedUser.isFederatedUser()) {
+            authenticatedUser.setFederatedIdPName(context.getExternalIdP().getIdPName());
+        } else {
+            /* User ID can only be set for local users. If the user is a federated user, will be ignored.
+             */
+            AbstractUserStoreManager userStoreManager = resolveUserStoreManager(context, userStore.getName());
+            authenticatedUser.setUserStoreDomain(userStore.getName());
+            String userId = user.resolveUserId(userStoreManager);
+            authenticatedUser.setUserId(userId);
+            authenticatedUser.setUserName(userStoreManager.getUserNameFromUserID(userId));
+        }
+        authenticatedUser.setAuthenticatedSubjectIdentifier(
+                user.resolveSubjectIdentifier(authenticatedUser.isFederatedUser()));
+        authenticatedUser.setTenantDomain(context.getTenantDomain());
+
+        return authenticatedUser;
+    }
+
+    /**
+     * Cast the given object to String.
+     * @param object    Object that need to cast.
+     *
+     * @return String which cast to the String class.
+     * @throws ActionExecutionResponseProcessorException If object cannot be cast to String.
+     */
+    private String castToString(Object object)
+            throws ActionExecutionResponseProcessorException {
+
+        try {
+            return objectMapper.convertValue(object, String.class);
+        } catch (IllegalArgumentException e) {
+            throw new ActionExecutionResponseProcessorException("The provided value cannot be cast to a string:" +
+                    object, e);
+        }
+    }
```

Comment:
we need to have facility to troubleshoot as input validations fails since this is related to an extension.

Reply:
Fixed with commit e35699ea008a4b1df7251ac6162b6e9634aacf77